### DISCUSSION TITLE
Using right Persister for Sub class mapping

### DISCUSSION THREAD

**Discussion by gkatoch** [nhibernate/nhibernate-core#2783] [Q&A]
Hi there ,
**We have defined an hbm for subclass like this:**
<img width="715" alt="Hbm mapping" src="https://user-images.githubusercontent.com/39444347/119004951-a34a4880-b9ac-11eb-8d2e-9f64493c3c08.PNG">

This creates an instance of SingleTableEntityPersister which has a table name of Topologicallink i.e. base class. This leads to failure in select count calls as properties of child class are not accessible from base tables.
On debugging we found that we always pick 1st qualified table name for SingleTableEntityPersister. Any idea why it has been done so ? (Can we override this to give persister of child class always)

We are not using JoinedSubclassEntityPersister as this was causing bottlenecks in lazy loading as any call to parent table makes a left outer join with child table (Even when the relation is from parent class only). Any way to avoid that ?

We are looking for a smarter way to determine if Table name is to be used for parent or subclass to avoid extra joins. Any idea how that can be done
Any help would be appreciated.
<img width="501" alt="Persisiter" src="https://user-images.githubusercontent.com/39444347/119004491-4058b180-b9ac-11eb-9279-ee59719dde9d.PNG">

**Comment by nishantgupta204** [comment]
I am also struggling to understand this reason why it is always picking up first qualified name every time. Can you someone help us answer this question ASAP?
